typescript-async-await/exceptions-best.ts

```diff
@@ -0,0 +1,38 @@
+import { read } from './read.js';
+
+// Keep track of app load time so each log message can be timed.
+// Log messages should all be approximately 1 second apart.
+const startTime = Date.now();
+const elapsed = (): string =>
+  `${Math.round((Date.now() - startTime) / 1000)}s -`;
+
+async function throwOnce(): Promise<void> {
+  const message = await read('foo', true);
+  console.log(elapsed(), 'throwOnce:', message);
+}
+
+async function throwSeveral(): Promise<void> {
+  let message = await read('foo1', true);
+  console.log(elapsed(), 'throwSeveral1:', message);
+  message = await read('foo2', true);
+  console.log(elapsed(), 'throwSeveral2:', message);
+  message = await read('foo3', true);
+  console.log(elapsed(), 'throwSeveral3:', message);
+}
+
+async function throwChained(): Promise<void> {
+  const message1 = await read('foo-chain', true);
+  console.log(elapsed(), 'throwChained1:', message1);
+  const message2 = await read(message1, false);
+  console.log(elapsed(), 'throwChained1:', message2);
+  const message3 = await read(message2, false);
+  console.log(elapsed(), 'throwChained1:', message3);
+}
+
+try {
+  await throwOnce()
+    .then(() => throwSeveral())
+    .then(() => throwChained());
```

Comment:
You want to refactor this to use async/await. You are awaiting `throwOnce` correctly, but you are chaining `throwSeveral` and `throwChained` onto your `throwOnce`. While yes, these would not be accessible if `throwOnce` rejects, you still don't want it to be dependent on another like this.